### DISCUSSION TITLE
Reading struct from contract, no keys in return data

### DISCUSSION THREAD

**Discussion by CryptoKira** [wevm/wagmi#2668] [Q&A]
I'm reading data from a contract, nothing unusual.
But when I get the return data, its missing the struct keys, and also has a letter 'n' at the end

For example, this is my return data, with no keys present: 

```
0: "0x0000000000000000000000000000000000000000"
1: 1688282366n
2: 6780862756357273079700164n
3: 100000000000000000000000n
4: 31572576933298532376928n
5: 6703581869116643316320430n
6: 1688282366n
```

Could it be related to this issue? https://github.com/wagmi-dev/wagmi/issues/460

**Comment by llllvvuu** [comment]
The `n` is intentional, it's native JavaScript `bigint` (`viem` ditched `BigNumber`).

The object -> arrays thing, I'm not sure if it's intentional but I noticed that behavior too and just switched my DApp to consume arrays.

**Comment by jxom** [comment]
https://viem.sh/docs/faq.html#why-is-a-contract-function-return-type-returning-an-array-instead-of-an-object

**Comment by CryptoKira** [comment]
Ok thanks.
